### DISCUSSION TITLE
[FEATURE] Cross-Realm - Battlegrounds and Arenas

### DISCUSSION THREAD

**Discussion by devovh** [azerothcore/azerothcore-wotlk#3926] [Ideas]
**Additional context**
Hello, a good function/module would be to introduce Cross-Realm(InterRealm) to the above-mentioned matters.

*Cross-Realm - Battlegrounds and Arenas

This module would rely on fact that having, for example, two realms, players from these realms could play together the individual cases listed above, i.e. two world servers connecting together on a separate world server.

**Comment by EricksOliveira** [comment]
Any information on that?

**Comment by Yehonal** [comment]
The only way to do it is to create a cluster architecture with a dedicated service to handle the BG system. It's needed that the system will be integrated directly inside the core, however, it's not in our schedule but would be nice to collect information on other research done about it here

**Comment by Open-Wow** [comment]
As I remember (I may be wrong) but Firestorm is already using an inter-realm or inter-server BG system.

**Comment by Yehonal** [comment]
in that case would be nice to understand "how"...since the expansions are completely different. To be honest, I don't think so.

But I know about projects that created it. The key is simple: **"money"**. If you have 10k online players and a VIP account, you can essentially pay a salary to 6-7 high-skilled C++ developers that in 1 or 2 years can create such a complex system for you...and sadly they won't share them since it's their core-business.

**Comment by EricksOliveira** [comment]
One of the servers using this type of service that comes to my attention is only Blizzard and Warmane.

**Comment by htc16** [comment]
That would be wonderfully good.

**Comment by devovh** [comment]
As for Inter-Realm, it can be used for e.g. merger of 2 separate projects in terms of some integration from this site. Cross World on a separate server where players from 2 separate server projects can connect. After all, when it comes to Cross, this function was used at Retail even in the times of Wotlk because it is about this addition and azerothcore is only for wotlk. And I come to the conclusion that this feature is blizzlike and yet in the emulation zone they pride themselves on being a blizzlike emulator.
